gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/FlowTriggerDecorator.java

```diff
@@ -116,54 +100,41 @@ public FlowTriggerHandler(Config config, Optional<MultiActiveLeaseArbiter> lease
    * @throws IOException
    */
   public void handleTriggerEvent(Properties jobProps, DagActionStore.DagAction flowAction, long eventTimeMillis,
-      boolean isReminderEvent, boolean skipFlowExecutionIdReplacement) throws IOException {
-    if (multiActiveLeaseArbiter.isPresent()) {
-      MultiActiveLeaseArbiter.LeaseAttemptStatus leaseAttemptStatus = multiActiveLeaseArbiter.get().tryAcquireLease(
-          flowAction, eventTimeMillis, isReminderEvent, skipFlowExecutionIdReplacement);
-      // The flow action contained in the`LeaseAttemptStatus` from the lease arbiter contains an updated flow execution
-      // id. From this point onwards, always use the newer version of the flow action to easily track the action through
-      // orchestration and execution.
+      boolean isReminderEvent, boolean skipFlowExecutionIdReplacement)
+      throws IOException {
+
+    try {
+      LeaseAttemptStatus leaseAttemptStatus = null;
+      leaseAttemptStatus =
+          this.tryAcquireLease(flowAction, eventTimeMillis, isReminderEvent, skipFlowExecutionIdReplacement);
       if (leaseAttemptStatus instanceof MultiActiveLeaseArbiter.LeaseObtainedStatus) {
-        if (isReminderEvent) {
-          this.leasesObtainedDueToReminderCount.mark();
+          MultiActiveLeaseArbiter.LeaseObtainedStatus leaseObtainedStatus = (MultiActiveLeaseArbiter.LeaseObtainedStatus)
+              leaseAttemptStatus;
+          if (persistFlowAction(leaseObtainedStatus)) {
```

Comment:
I believe at one point, originally, this terminology of "flow action" made sense to me, but as we've gone deeper to define a bevy of new actions pretty exclusively around the name `DagAction`, I'm left to wonder:
* what is a "flow action"?
* how is it different from a DagAction?
* could the two concepts be consolidated into one, to also unify the naming/terminology?